test/unit/client_test.coffee

```diff
@@ -107,6 +108,12 @@ describe "Client", ->
         error = _err
       return error
 
+    it "returns promise and resolves it", ->
+      promise = client.push(exception)
```

Comment:
> es6-promise is 17kb minified. Too big for us.

That is indeed a bit big, but only incompatible browser would incur that hit yes? I'd still prefer it to maintaining our own. If that is not acceptable to you, then I think we should use a callback.

@zefer do you have any thoughts on this?